### DISCUSSION TITLE
part-ii/2009-28/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#150] [Question comments]
Part II 2009 Coding and Cryptography: Paper 2, Section I, $4 \mathrm{H}$ 

https://questions.tripos.org/part-ii/2009-28/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

Note that $25 = 5^2$. Hence $2886^2 \equiv 5^2 \pmod{3953}$, so $3953$ divides 

$$2886^2 - 5^2 = (2886 - 5)(2886 + 5) = 2881 \cdot 2891$$

We compute $\gcd(3953,2881)$ and hope that this gives us a non-trivial factor of $3953$. We use Euclid's algorithm.

$$
3953 = 1 \cdot 2881 + 1072 \\
2881 = 2 \cdot 1072 + 737 \\
1072 = 1 \cdot 737 + 335 \\
737 = 2 \cdot 335 + 67 \\
335 = 5 \cdot 67
$$

Hence $67$ divides $3953$, and we have that $\frac{3953}{67} = 59$. Therefore $(p,q)$ is $(59,67)$ (in some order).

</details>
